taroscript/send.go

```diff
@@ -153,6 +154,107 @@ func PrepareOutputAssets(input *taropsbt.VInput,
 	return nil
 }
 
+// SignVirtualTransaction updates the new asset (the root asset located at the
+// change output in case of a non-interactive or partial amount send or the
+// full asset in case of an interactive full amount send) by creating a
+// signature over the asset transfer, verifying the transfer with the Taro VM,
+// and attaching that signature to the new Asset.
+func SignVirtualTransaction(input *taropsbt.VInput, outputs []*taropsbt.VOutput,
+	signer Signer, validator TxValidator) error {
+
+	prevAssets := commitment.InputSet{
+		input.PrevID: input.Asset(),
+	}
+	newAsset := outputs[1].Asset
```

Comment:
Seems to assume a strict ordering re the outputs here?

Reply:
Yes, was related to the "convention" I was initially going with. But now the logic only looks at the `IsChange` and `IsInteractive` flags and doesn't assume a specific location for things. In this case the first output will always be the one with the signature, since either it's the change output (or tombstone) in case of a split, or it's the new asset in case of an interactive full value send.